base/applications/mspaint/toolsmodel.h

```diff
@@ -30,6 +30,31 @@ enum TOOLTYPE
 
 /* CLASSES **********************************************************/
 
+struct ToolBase
+{
+    TOOLTYPE m_tool;
+    HDC m_hdc;
+    COLORREF m_fg, m_bg;
+
+    ToolBase(TOOLTYPE tool) : m_tool(tool), m_hdc(NULL), m_fg(0), m_bg(0)
+    {
+    }
+
+    virtual ~ToolBase()
+    {
+    }
+
+    virtual void OnDown(BOOL bLeftButton, LONG x, LONG y, BOOL bDoubleClick);
```

Comment:
OnDown/Move/Up doesn't sound specific: what is going down/up/moving?

Reply:
Renamed as `OnButtonDown/OnMouseMove/OnButtonUp`.